### DISCUSSION TITLE
Barcamp mode

### DISCUSSION THREAD

**Discussion by rixx** [pretalx/pretalx#1072] [Ideas]
This issue collects the various smaller items to be completed for a barcamp mode.

- [ ] A flag (not easily editable) making an event a conference or a barcamp.
- [ ] Add TalkSlots for submissions directly
- [ ] Disable accept-review-confirm cycle entirely
- [ ] Expose scheduling API endpoint
- [ ] Expose scheduling page
- [ ] Provide a periodic (~5 minutes) schedule export (schedule.xml etc) of the WIP version

**Comment by saschafoerster** [comment]
I think a starter would make it possible to quickly digitalize the sessions, so that one member of the teams quickly copies the sessions. That is what we most of the time do during a Barcamp, so that a digital version is available at least after the first session. This would make it really easy to share and most of all to document all the sessions of all barcamps in a nice tool.

**Comment by rixx** [comment]
Since the feature of adding breaks was added, we are a bit closer, in that we can now add talk slots directly from the schedule editor.

Apart from making those usable for actual talks and not just breaks, the current interface works via dragging a selection, which seems problematic for mobile devices. Since I imagine those would be predominant at barcamps, that would be another challenge in implementing this.

**Comment by saschafoerster** [comment]
I am still dependent on docker, so I hope the community can soon push their update to version 1.0.4. and I can start testing. :) https://github.com/pretalx/pretalx-docker/pull/15

**Comment by saschafoerster** [comment]
> the current interface works via dragging a selection, which seems problematic for mobile devices. Since I imagine those would be predominant at barcamps, that would be another challenge in implementing this.

Most of times we use laptops, even at Barcamps, because it is much quicker for typing. Some drag handles in mobile situations would be nice as well. They could be turned off on desktop situations (and then normal drag&drop) and then turned on in mobile situations, where only the drag-handles can be used to drag and drop, the rest of the area stays usable for scrolling.

**Comment by rixx** [comment]
Please note that we can only add break slots directly from the schedule editor. A barcamp mode is still very far from being implemented. (Also, please let's keep other issues in their respective repositories. I suspect the discussion on this issue will get long and drag on over the coming years, so keeping it to the point is helpful to everybody.)

**Comment by saschafoerster** [comment]
I recently found your plugin https://github.com/pretalx/pretalx-public-voting. In the actual situation we would like to transfer our local Barcamps to Venueless http://github.com/venueless/venueless. Creating digital Barcamps might be a bit different from having local Barcamps, but to have a public voting seems to me like the best "minimal viable product" to get a little step closer to an full barcamp mode. We maybe would not put the voting at the same day of the Barcamp, but have a public voting phase some days before the actual Barcamp. I just wanted to encourage somehow the further integration of the public vote feature as part of the bigger Barcamp mode and help other finding this resources.
